spec/06-expressions.md

```diff
@@ -245,7 +245,11 @@ ArgumentExprs ::=  ‘(’ [Exprs] ‘)’
 Exprs         ::=  Expr {‘,’ Expr}
 ```
 
-An application `´f(e_1 , \ldots , e_m)´` applies the function `´f´` to the argument expressions `´e_1, \ldots , e_m´`. For this expression to be well-typed, the function must be *applicable* to its arguments, which is defined next by case analysis on ´f´'s type.
+An application applies some function `´f´` to its arguments. It either takes the
+form `´f(e_1 , \ldots , e_m)´` where `´e_1, \ldots , e_m´` are argument
+expressions, or the form `´f e_1` where `´e_1´` is a single block expression.
```

Comment:
I'd suggest adding it after sequence arg `: _*` explanation.

If the function takes a single parameter, the argument may be supplied as a block expression.

If all parameters but one have defaults, can I use block expr syntax?

Someone might also assume the block expr supplies a tuple of args.

In a separate paragraph, there is room for such further clarification.

Reply:
Yes, it's much better in terms of a single supplied/required argument. I liked supplied better because that's the form we're currently looking at and well-typedness has been dealt with above. One could still argue it suddenly declares all single block expressions valid even if there are more formal parameters or it's not well typed but you'd have to be arguing from pretty bad faith to get there I think.